syntax = "proto3";

package envoy.config.trace.v3;

import "google/protobuf/any.proto";

import "udpa/annotations/status.proto";
import "udpa/annotations/versioning.proto";
import "validate/validate.proto";

option java_package = "io.envoyproxy.envoy.config.trace.v3";
option java_outer_classname = "HttpTracerProto";
option java_multiple_files = true;
option (udpa.annotations.file_status).package_version_status = ACTIVE;

// [#protodoc-title: Tracing]
// Tracing :ref:`architecture overview <arch_overview_tracing>`.

// The tracing configuration specifies settings for an HTTP tracer provider used by Envoy.
//
// Envoy may support other tracers in the future, but right now the HTTP tracer is the only one
// supported.
//
// .. attention::
//
//   Use of this message type has been deprecated in favor of direct use of
//   :ref:`Tracing.Http <envoy_v3_api_msg_config.trace.v3.Tracing.Http>`.
message Tracing {
  option (udpa.annotations.versioning).previous_message_type = "envoy.config.trace.v2.Tracing";

  // Configuration for an HTTP tracer provider used by Envoy.
  //
  // The configuration is defined by the
  // :ref:`HttpConnectionManager.Tracing <envoy_v3_api_msg_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.Tracing>`
  // :ref:`provider <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.Tracing.provider>`
  // field.
  message Http {
    option (udpa.annotations.versioning).previous_message_type =
        "envoy.config.trace.v2.Tracing.Http";

    reserved 2;

    reserved "config";

    // The name of the HTTP trace driver to instantiate. The name must match a
    // supported HTTP trace driver.
    // See the :ref:`extensions listed in typed_config below <extension_category_envoy.tracers>` for the default list of the HTTP trace driver.
    string name = 1 [(validate.rules).string = {min_len: 1}];

    // Trace driver specific configuration which must be set according to the driver being instantiated.
    // [#extension-category: envoy.tracers]
    oneof config_type {
      google.protobuf.Any typed_config = 3;
    }
  }

  // Provides configuration for the HTTP tracer.
  Http http = 1;
}
